homeassistant/helpers/script.py

```diff
@@ -582,47 +684,60 @@ def __init__(
         sequence: Sequence[Dict[str, Any]],
         name: Optional[str] = None,
         change_listener: Optional[Callable[..., Any]] = None,
-        if_running: Optional[str] = None,
-        run_mode: Optional[str] = None,
+        script_mode: str = DEFAULT_SCRIPT_MODE,
+        queue_max: int = DEFAULT_QUEUE_MAX,
         logger: Optional[logging.Logger] = None,
         log_exceptions: bool = True,
     ) -> None:
         """Initialize the script."""
-        self._logger = logger or logging.getLogger(__name__)
         self._hass = hass
         self.sequence = sequence
         template.attach(hass, self.sequence)
         self.name = name
         self._change_listener = change_listener
+        self._script_mode = script_mode
+        if logger:
```

Comment:
All this logic, I feel like we should just make it required to pass in a logger. Let the instantiater take care of this. Our logs are not very helpful by making them relative to `helpers.script`.

Reply:
I agree, but how do I do that without forcing all existing users to provide a logger (i.e., without adding it as a positional parameter)? I felt like that would have been a more disruptive change than just providing a reasonably default logger.

Reply:
There are only 2 users right now ?

Reply:
17:
```
(venv) phil@GGHWFS1:~/repos/home-assistant (script-modes)$ grep -m 1 'Script(' homeassistant -r --include='*.py'
homeassistant/helpers/script.py:class _StopScript(Exception):
homeassistant/components/philips_js/media_player.py:    on_script = Script(hass, turn_on_action) if turn_on_action else None
homeassistant/components/lg_netcast/media_player.py:    on_action_script = Script(hass, on_action) if on_action else None
homeassistant/components/knx/__init__.py:        self.script = Script(hass, action, script_name)
homeassistant/components/kodi/media_player.py:            turn_on_action = script.Script(
homeassistant/components/wake_on_lan/switch.py:        self._off_script = Script(hass, off_action) if off_action else None
homeassistant/components/template/fan.py:        self._on_script = Script(hass, on_action)
homeassistant/components/template/lock.py:        self._command_lock = Script(hass, command_lock)
homeassistant/components/template/alarm_control_panel.py:            self._disarm_script = Script(hass, disarm_action)
homeassistant/components/template/light.py:        self._on_script = Script(hass, on_action)
homeassistant/components/template/cover.py:            self._open_script = Script(hass, open_action)
homeassistant/components/template/vacuum.py:        self._start_script = Script(hass, start_action)
homeassistant/components/template/switch.py:        self._on_script = Script(hass, on_action)
homeassistant/components/samsungtv/media_player.py:        on_script = Script(hass, turn_on_action)
homeassistant/components/automation/__init__.py:            action_script = script.Script(
homeassistant/components/webostv/media_player.py:    on_script = Script(hass, turn_on_action) if turn_on_action else None
homeassistant/components/intent_script/__init__.py:            conf[CONF_ACTION] = script.Script(
homeassistant/components/script/__init__.py:        self.script = Script(
```